### PR TITLE
Support HiDPI on Linux by querying the Xft DPI through XSETTINGS.

### PR DESCRIPTION
**Pull request by pcwalton**
The Xft DPI is set when the user adjusts the scale in the Ubuntu
"Displays" preference pane.

This depends on `rust-xsettings`, which in turn uses the C library
`libxsettings-client-0`. Unfortunately, this library is not installed by
default on most Linux distributions. To avoid unnecessary dependencies,
this commit places support for HiDPI on Linux behind the `use-xsettings`
Cargo feature.

### CODE REVIEW

**Comment by pcwalton** [comment]
I'm happy to adjust how the feature works if you'd like.

File: Cargo.toml [hunk 1]
```diff
@@ -86,6 +90,10 @@ wayland-kbd = "0.3.3"
 wayland-window = "0.2.2"
 x11-dl = "~2.3"
 
+[target.i686-unknown-linux-gnu.dependencies.xsettings]
```
Comment (SimonSapin) [inline Cargo.toml]: Nit: could you format this as `xsettings = {…}` in the existing `[target.….dependencies]` sections?

**Comment by tomaka** [comment]
I've been trying to use more and more dynamic loading in glutin. X11 is currently loaded with `dlopen`, because systems that have wayland installed and not X11 should be able to run glutin apps as well. And #605 is an old PR that was supposed to make all the additional X libraries optional, notably because raspbian (debian for the raspberry pi) doesn't have them.

I don't know what the state of art about this is, but if you distribute pre-compiled binaries you don't want to distribute a X11 version, a wayland version, a X11-with-xcursor version, and so on..

It would be nicer if `x-settings` followed the same principle instead of using Cargo features.

**Comment by ishitatsuyuki** [comment]
@pcwalton If you're still interested in this, this should now be added to winit. There's a non-standard, actual data based HiDPI detection currently.

**Comment by francesca64** [comment]
Relevant winit issues: tomaka/winit#105 tomaka/winit#169

**Pull request closed without merging**